src/test/java/com/hazelcast/eurekast/one/EurekastOneDiscoveryStrategyTest.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    @Test
+    public void shouldOnlyDiscoverInstancesWithUpStatus(){
+        Application application = new Application();
+        List<InstanceInfo> infoList = InstanceInfoGenerator
+                .newBuilder(5, 1)
+                .build()
+                .toInstanceList();
+        for (InstanceInfo info : infoList) {
+            info.setStatus(InstanceInfo.InstanceStatus.DOWN);
+            application.addInstance(info);
+        }
+
+        when(eurekaClient.getApplication(APPLICATION_NAME)).thenReturn(application);
+
+        Iterable<DiscoveryNode> nodes = strategy.discoverNodes();
+        assertThat(nodes.iterator().hasNext(), is(false));
+    }
+
+    @Test
+    public void shouldOnlyDiscoverInstancesWithValidAddress(){
+        Application application = new Application();
+        InstanceInfo mockInfo = mock(InstanceInfo.class);
+        when(mockInfo.getId()).thenReturn(RandomStringUtils.random(42));
+        when(mockInfo.getStatus()).thenReturn(InstanceInfo.InstanceStatus.UP);
+        when(mockInfo.getIPAddr()).thenReturn("local");
+
+        application.addInstance(mockInfo);
+
+        when(eurekaClient.getApplication(APPLICATION_NAME)).thenReturn(application);
+
+        Iterable<DiscoveryNode> nodes = strategy.discoverNodes();
+
+        verify(eurekaClient).getApplication(APPLICATION_NAME);
+        verify(mockInfo).getStatus();
+        verify(mockInfo).getIPAddr();
+
+        assertThat(nodes.iterator().hasNext(), is(false));
+    }
+
+    @Test
+    public void shouldContinueVerifyRegistrationWhenEurekaClientThrowException() {
+        when(eurekaClient.getApplication(APPLICATION_NAME))
+                .thenThrow(new RuntimeException())
+                .thenReturn(new Application());
+
+        strategy.verifyEurekaRegistration();
+        verify(eurekaClient, times(2)).getApplication(eq(APPLICATION_NAME));
+    }
+
+    @Test
+    public void shouldRetryNodeDiscoveryUntilLimitReached(){
+        when(eurekaClient.getApplication(APPLICATION_NAME))
+                .thenReturn(null);
+
+        Iterable<DiscoveryNode> actual = strategy.discoverNodes();
+        verify(eurekaClient, times(NUM_RETRIES)).getApplication(APPLICATION_NAME);
+
+        assertThat(actual.iterator().hasNext(), is(false));
+    }
+
+
+    @Test
+    public void shouldEscalateWhenErrorCaughtInVerification(){
+        expectedException.expect(Error.class);
+        when(eurekaClient.getApplication(APPLICATION_NAME))
+                .thenThrow(new Error());
+
+        strategy.start();
+        verify(applicationInfoManager).setInstanceStatus(InstanceInfo.InstanceStatus.UP);
+    }
+
+    @Test
+    public void shouldNotRegisterWhenClientModeEnabled(){
+        when(eurekaClient.getApplication(APPLICATION_NAME))
+                .thenReturn(new Application());
+        reset(applicationInfoManager);
+
+        strategy.setClientMode(true);
```

Comment:
I think this should be a separate test class, to remove the final from clientMode just to put it into the same testcase is not too nice :)